File: day7/recursion.py
class Recursion:
    def __init__(self):
        pass

    def factorialWithRecursion(self, value:int ) -> int:
        if value == 0:
            return 1

        return value * self.factorialWithRecursion(value-1)


recursion = Recursion()
result = recursion.factorialWithRecursion(5)
print(result)            

File: day7/recursion.js
class Recursion {

    factorialWithRecursion(n) {
        if (n == 0) {
            return 1;
        }

        return n * this.factorialWithRecursion(n - 1);
    }

    factorialWithoutRecursion(n) {
        let factorial = 1;
        for (let v = n; v >= 1; v--) {
            factorial *= v;
        }
        return factorial;
    }
}

const obj = new Recursion();
result = obj.factorialWithRecursion(9);
console.log(result);
result = obj.factorialWithoutRecursion(9);
console.log(result);